### DISCUSSION TITLE
Elastic Defend

### DISCUSSION THREAD

**Discussion by kiaraho** [Security-Onion-Solutions/securityonion#10095] [2.4] [answered]
Planning a test of 2.4 in a standalone deployment and wanted to inquire if anybody has successfully added the Elastic Defend integration. If so, are there any integration steps that are different or unique to the Security Onion implementation of Fleet?

**Comment by InfosecGoon** [comment, marked as answer]
It appears functional:

https://twitter.com/Cole_Van_/status/1643430005819547649

**Comment by oestreicher42** [comment]
I tried, but was unsuccesful.
Kibana gives me the following error message

--
Error installing endpoint 8.7.1: status_exception Root causes: status_exception: Transform requires the transform node role for at least 1 node, found no transform nodes
--

I tried to add the transform role to my search node in the minion pillar:

--
elasticsearch:
  node_roles:
    - data
    - ingest
    - transform
--

But this had no effect. I h<d not time to troubleshoot this further, but perhaps the missing transform role is a known problem.

Tried with beta 2 on rocky 9.1 in a distributed setup.

**Comment by EchoGangster** [comment]
I have tried to install it a few times and every time I do the in Fleet the Agent goes unhealthy.  The cause of the unhealthy agent is the Elastic Defend integration.  Error states "Policy response failure

The Endpoint did not apply the Policy correctly. Expand the Policy response above for more details."

**Reply by defensivedepth** [reply]
@EchoGangster Please start a new discussion.
